components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -3263,6 +3263,7 @@ public void deleteAPI(String apiUuid, String organization) throws APIManagementE
                     "the recommendation system. "
             );
             isError = true;
+            throw e;
```

Comment:
Do we have a requirement to throw this?
The flow we have designed here is to try all the delete steps despite of the failures. ( eg: 
```
try { 
    some delete step()
} catch (failure) {
    log()
}

try { 
    next delete step()
} catch (failure) {
    log()
}

etc
```

Reply:
If it's unable to find the relevant API  , do we need to execute next steps ?? can't we ignore the next executions and throw the error ?

Reply:
Fixed via : https://github.com/wso2/carbon-apimgt/pull/11114/commits/e41d31f1ee471c3d908a363cc5493708e6f46f26